oscilloscope/oscilloscope_server.py

```diff
@@ -0,0 +1,227 @@
+        """Get or set the vertical position.
+
+        Args:
+            channel (int): Which channel to get/set.
+            position (float): Vertical position in units of divisions. If None,
+                (the default), then we only query.
+
+        Returns:
+            (float): Vertical position in units of divisions.
+        """
+        dev = self.selectedDevice(c)
+        result = yield dev.vert_position(channel, position)
+        returnValue(result)
+
+    # HORIZONTAL
+
+    @setting(30, scale='v[s]', returns='v[s]')
+    def horiz_scale(self, c, scale=None):
+        """Set or query the horizontal scale.
+
+        Args:
+            scale (Value[s]): Horizontal scale, i.e. time per division. If None,
+                (the default), then we just query.
+
+        Returns:
+            (Value[s]): The horizontal scale.
+        """
+        dev = self.selectedDevice(c)
+        result = yield dev.horiz_scale(scale)
+        returnValue(result)
+
+    @setting(31, position='v[]', returns='v[]')
+    def horiz_position(self, c, position=None):
+        """Set or query the horizontal position.
+
+        Args:
+            position (float): Horizontal position in units of division.
+
+        Returns:
+            (float): The horizontal position in units of divisions.
+        """
+        dev = self.selectedDevice(c)
+        result = yield dev.horiz_position(position)
+        returnValue(result)
+
+    # TRIGGER
+
+    @setting(71, source=['s'], returns='s')
+    def trigger_source(self, c, source=None):
+        """Set or query trigger source.
+
+        Args:
+            source (str): 'EXT', 'LINE', 'CHANX' where X is channel number. If
+                None (the default) then we just query.
+
+        Returns:
+            (str): Trigger source.
+        """
+        dev = self.selectedDevice(c)
+        resp = yield dev.trigger_source(source)
+        returnValue(resp)
+
+    @setting(72, slope='s', returns='s')
+    def trigger_slope(self, c, slope=None):
+        """Set or query trigger slope.
+
+        Args:
+            slope (str): Trigger slope. If None, the default, we just query.
+                Allowed values are 'POS' and 'NEG'.
+
+        Returns:
+            (str): The trigger slope.
+        """
+        dev = self.selectedDevice(c)
+        resp = yield dev.trigger_slope(slope)
+        returnValue(resp)
+
+    @setting(73, level='v[V]', returns='v[V]')
+    def trigger_level(self, c, level=None):
+        """Set or query the trigger level.
+
+        Args:
+            level (Value[V]): Trigger level. If None (the default), we just
+                query.
+
+        Returns:
+            (Value[V]): The trigger level.
+        """
+        dev = self.selectedDevice(c)
+        resp = yield dev.trigger_level(level)
+        returnValue(resp)
+
+    # ACQUISITION
+
+    @setting(60, channel='i', returns='*v[s]*v[V]')
+    def get_trace(self, c, channel):
+        """Get a trace for a single channel.
```

Comment:
> Is the idea that this downloads a previously-captured trace, or does it capture and download?

The "idea" here (if you can call it that) was to replicate how the previous scope servers work so I can do work. We should set up nice interfaces in the future, and maybe we should rename this one to make room for future improvements, but I'm not at all interested in adding new functionality now.

> I don't really see any settings here for doing things like triggering a single-shot capture; do you plan to support that sort of thing in a generic way?